apple/OmnivoreKit/Sources/App/Views/WebReader/WebReader.swift

```diff
@@ -84,11 +85,19 @@ struct WebReader: PlatformViewRepresentable {
   private func updatePlatformView(_ webView: WKWebView, context: Context) {
     if annotationSaveTransactionID != context.coordinator.lastSavedAnnotationID {
       context.coordinator.lastSavedAnnotationID = annotationSaveTransactionID
-      (webView as? OmnivoreWebView)?.dispatchEvent(.saveAnnotation(annotation: annotation))
+      do {
+        try (webView as? OmnivoreWebView)?.dispatchEvent(.saveAnnotation(annotation: annotation))
```

Comment:
Does the dispatchEvent call sometime fail?

Reply:
@satindar I think technically it can throw from JS. I haven't been able to force it, but we were kind of silently failing before.

Reply:
Well, I did make it so generating the event can throw if we can't serialize the event data.

Reply:
Nice. Should be more robust now 👍🏽